File: #905_sortArrayByParity.py
class Solution:
    def sortArrayByParity(self, A: List[int]) -> List[int]:
        i = 0
        j = len(A)-1
        while i<j:
            if A[i]%2 !=0:
                if A[j]%2 == 0:
                    A[i],A[j] = A[j],A[i]
                    i+=1
                    j-=1
                else:
                    j-=1
            else:
                i+=1
        return A
        

File: Java/#905_sortArrayByParity.java
class Solution {
    public int[] sortArrayByParity(int[] A) {
        int i=0;
        int j=A.length-1;
        while(i<j){
            if(A[i]%2!=0){
                if(A[j]%2==0){
                    int t=A[i];
                    A[i]=A[j];
                    A[j]=t;
                }else{
                    j--;
                }
            }else{
                i++;
            }
        }
        return A;
    }
}
